addon/index.d.ts

```diff
@@ -0,0 +1,759 @@
+   */
+  readonly isFinished: boolean;
+
+  /**
+   * True if the task is still running.
+   */
+  readonly isRunning: boolean;
+
+  /**
+   * Describes the state that the task instance is in. Can be used for debugging,
+   * or potentially driving some UI state. Possible values are:
+   *
+   * - `"dropped"`: task instance was canceled before it started
+   * - `"canceled"`: task instance was canceled before it could finish
+   * - `"finished"`: task instance ran to completion (even if an exception was thrown)
+   * - `"running"`: task instance is currently running (returns true even if
+   *     is paused on a yielded promise)
+   * - `"waiting"`: task instance hasn't begun running yet (usually
+   *     because the task is using the {@linkcode TaskProperty#enqueue .enqueue()}
+   *     task modifier)
+   *
+   * The animated timeline examples on the [Task Concurrency](/#/docs/task-concurrency)
+   * docs page make use of this property.
+   */
+  readonly state: 'dropped' | 'canceled' | 'finished' | 'running' | 'waiting';
+
+  /**
+   * True if the TaskInstance was canceled before it could
+   * ever start running. For example, calling
+   * {@linkcode Task#perform .perform()} twice on a
+   * task with the {@linkcode TaskProperty#drop .drop()} modifier applied
+   * will result in the second task instance being dropped.
+   */
+  readonly isDropped: boolean;
+
+  /**
+   * Cancels the task instance. Has no effect if the task instance has
+   * already been canceled or has already finished running.
+   *
+   * @param cancelReason Defaults to `".cancel() was explicitly called"`.
+   */
+  cancel(cancelReason?: string): void;
+
+  /**
+   * Returns a promise that resolves with the value returned
+   * from the task's (generator) function, or rejects with
+   * either the exception thrown from the task function, or
+   * an error with a `.name` property with value `"TaskCancelation"`.
+   */
+  then<TResult1 = T, TResult2 = never>(
+    onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | null,
+    onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | null
+  ): Promise<TResult1 | TResult2>;
+
+  catch<TResult = never>(
+    onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | null
+  ): Promise<T | TResult>;
+
+  finally(onfinally?: (() => void) | null): Promise<T>;
+}
+
+/**
+ * A {@link TaskProperty} is the Computed Property-like object returned
+ * from the {@linkcode task} function. You can call Task Modifier methods
+ * on this object to configure the behavior of the {@link Task}.
+ *
+ * See [Managing Task Concurrency](/#/docs/task-concurrency) for an
+ * overview of all the different task modifiers you can use and how
+ * they impact automatic cancelation / enqueueing of task instances.
+ */
+export interface TaskProperty<T = unknown, Args extends any[] = []> extends ComputedProperty<Task<T, Args>> {
+  volatile: never;
+  readOnly: never;
+  property: never;
+  meta: never;
```

Comment:
This extends `ComputedProperty<Task<T, Args>>` because that seems to be the only public way to make `.get` work. However, the CP properties doesn't actually exist on a `TaskProperty` (not on recent Ember versions anyway), hence retyping to `never`.

Reply:
@chancancode can you file a types issue referencing this at https://github.com/typed-ember/ember-cli-typescript? Now that we're landing the bump in the core types to 3.7, and with some other work I have in mind to tackle in the next few months, we *should* be able to make `get` work correctly. (Conditional types!)